upload/catalog/model/smailyforopencart/helper.php

```diff
@@ -210,7 +210,7 @@ public function addSentCart($customer_id) {
      */
     public function getSyncTime() {
         $this->load->model('setting/setting');
-        $sync_time = $this->model_setting_setting->getSettingValue('smaily_for_opencart_sync_time') ?: date('c', 0);
+        $sync_time = $this->model_setting_setting->getSetting('smaily_for_opencart_sync_time') ?: date('c', 0);
```

Comment:
I think it is not going to work, because `ControllerSmailyForOpencartCronCustomers` controller will set `smaily_for_opencart_sync_time` setting value in `smaily_for_opencart` namespace.

If catalog `setting/setting` model does not provide necessary functions to read from a namespaced setting value, then I think we need to extend our catalog helper model and add `getSettingValue` to it.

Reply:
Yes, I have now looked more into it and you are correct, the current solution will not work. `getSetting` function will get the whole array of settings under `smaily_for_opencart` code.
I think there is no need to change behavior for getting settings because you can always check if the key you are looking for exist in the setting array. 

```
if (array_key_exists('smaily_for_opencart_sync_time', $settings)) {
     $sync_time = $settings['smaily_for_opencart'];
}
```

But the real issue is that the `editSettingValue` will not create a new instance, it will only update so the code block for adding update time will not work.
https://github.com/sendsmaily/smaily-opencart-module/blob/c84e2f5b20f2df66f34d161d3f20087049ba838a/upload/catalog/controller/smailyforopencart/cron_customers.php#L77-L82

https://github.com/sendsmaily/smaily-opencart-module/blob/c84e2f5b20f2df66f34d161d3f20087049ba838a/upload/catalog/model/smailyforopencart/helper.php#L217-L222

Also, I think the settings page will overwrite the current `sync_time` when settings are saved on admin page. https://github.com/sendsmaily/smaily-opencart-module/blob/c84e2f5b20f2df66f34d161d3f20087049ba838a/upload/admin/controller/module/smaily_for_opencart.php#L50-L60

Reply:
So as a solution the sync time should be created when saving admin page settings for the first time and should be "overwritten" with the current time as the credentials are currently "overwritten". (actually this should be also changed in a further update as it was done in WooCommerce plugin)

Reply:
Yup. There is something funky going on with the setting values. I think we need to revise it and possibly might need to release 2.0 version of the module, before we can move ahead with this PR.

Reply:
I have introduced the solution described in https://github.com/sendsmaily/smaily-opencart-module/pull/46/commits/aa317fe46907ba65c370cacdebfe09c2fba59896